src/App.jsx

```diff
@@ -14,33 +17,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABET = 'Sort alphabetically';
+const SORT_LENGTH = 'Sort by length';
+const REVERSE = 'Reverse';
+const RESET = 'Reset';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function styleClass(btn) {
+  switch (btn) {
+    case SORT_ALPHABET:
+      return 'is-info';
+    case SORT_LENGTH:
+      return 'is-success';
+    case REVERSE:
+      return 'is-warning';
+    case RESET:
+      return 'is-danger';
+    default:
+      return '';
+  }
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function sortGoods(goods, { typeField, isReversed }) {
+  const changedGoods = [...goods];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (typeField) {
+    if (typeField === SORT_ALPHABET) {
+      if (isReversed) {
+        return changedGoods.sort((a, b) => a.localeCompare(b)).reverse();
+      }
+
+      return changedGoods.sort((a, b) => a.localeCompare(b));
+    }
+
+    if (typeField === SORT_LENGTH) {
+      if (isReversed) return changedGoods.sort((a, b) => b.length - a.length);
+
+      return changedGoods.sort((a, b) => b.length - a.length).reverse();
```

Comment:
```suggestion
    switch (typeField) {
      case SORT_ALPHABET:
        changedGoods.sort((a, b) => a.localeCompare(b));
        break;
      case SORT_LENGTH:
        changedGoods.sort((a, b) => b.length - a.length);
        break;
      default:
        throw new Error('Invalid sort type');
    }
```
- Use switch case instead
- Reverse is a separate operation and you do it later, so now need to combine so many conditions in the same place